CHANGELOG.md

```diff
@@ -7,6 +7,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 ## [Unreleased]
 ### Changed
 - More detailed and sophisticated search space user guide
+- Better human readable `__str__` representation of objective and targets
```

Comment:
Looking at the commit history, I think it would be great if you could combine some of the commits ("Fix Typo").

Reply:
I actually kept it as the last step before merging. But as I have you review I will do it now 👍

Reply:
Then feel free to resolve this once you have done that :)